### PR TITLE
Derek/concat

### PR DESCRIPTION
**Pull request by betaBison**
Replaces `navdata.add` with `navdata.concat` that allows merging `navdata` instances by either row or column

### CODE REVIEW

File: tests/parsers/test_navdata.py [hunk 1]
```diff
@@ -1284,19 +1290,123 @@ def test_add_pandas_df(df_simple, add_df):
...
+
+    Parameters
+    ----------
+    df_simple : pd.DataFrame
+        Simple pd.DataFrame with which to initialize NavData.
+
+    """
+
+    navdata_1 = NavData(pandas_df=df_simple)
+    navdata_2 = navdata_1.copy()
```
Comment (sriramyab) [inline tests/parsers/test_navdata.py:1319]: maybe you can also add tests on what happens if an completely empty navdata class is given as input? the concatenated one should be same as input one, while still maintaining the navdata class instance.

**Review by sriramyab — CHANGES_REQUESTED** [review]
Looks good to me. Minor suggestions/comments.

File: gnss_lib_py/parsers/navdata.py [hunk 2]
```diff
@@ -116,50 +116,104 @@ def from_numpy_array(self, numpy_array):
...
+
         self._build_navdata()
 
+        numpy_array = np.atleast_2d(numpy_array)
         for row_num in range(numpy_array.shape[0]):
             self[str(row_num)] = numpy_array[row_num,:]
 
-    def add(self, csv_path=None, pandas_df=None, numpy_array=None):
-        """Add new timesteps to existing array
+    def concat(self, navdata=None, axis=1, inplace=False):
```
[the remark this answers is not in the document]
Comment (betaBison) [reply gnss_lib_py/parsers/navdata.py:130]: Right, so it didn't make sense to duplicate functionality that already existed with our NavData initialization so now `concat` directly concatenates two NavData instances and it's up to the user to create the second NavData instance from csv, DataFrame or array. I added additional documentation in the NavData initialization of each option.

File: gnss_lib_py/parsers/navdata.py [hunk 3]
```diff
@@ -116,50 +116,104 @@ def from_numpy_array(self, numpy_array):
...
-                pandas_df = pd.read_csv(csv_path)
-        if pandas_df is not None:
-            if old_row_num == 0:
-                self.from_pandas_df(pandas_df)
-            else:
-                self.array = np.hstack((self.array, np.empty(pandas_df.shape).T))
-                for col in pandas_df.columns:
-                    self[col, new_data_cols] = np.asarray(pandas_df[col].values)
+
+        if not isinstance(navdata,NavData):
```
[the remark this answers is not in the document]
Comment (betaBison) [reply gnss_lib_py/parsers/navdata.py:166]: changed to "concat input data"

File: tests/parsers/test_android.py [hunk 4]
```diff
@@ -291,6 +291,37 @@ def test_get_and_set_str(derived):
...
+        pd.DataFrame for testing measurements
+    """
+
+    # remove first timestamp to match
+    pd_df = pd_df[pd_df['millisSinceGpsEpoch'] != pd_df.loc[0,'millisSinceGpsEpoch']]
+
+    # extract and combine gps and glonass data
+    gps_data = derived.where("gnss_id","gps")
+    glonass_data = derived.where("gnss_id","glonass")
+    combined_navdata = gps_data.concat(glonass_data)
```
[the remark this answers is not in the document]
Comment (betaBison) [reply tests/parsers/test_android.py:311]: Right. Order does matter. A concatenated to B is different than B concatenated to A

[the remark this answers is not in the document]
Comment (betaBison) [reply tests/parsers/test_android.py:311]: Added comparison of alternate of glonass + gps as suggested. That ordering difference was already tested in `test_concat` but doesn't hurt to add another for assurance since that test doesn't take long.

File: tests/parsers/test_navdata.py (hunk 1 above)
Comment (betaBison) [reply tests/parsers/test_navdata.py:1319]: Added suggested test with empty Navdata. The test should fail if `axis=0` because the number of columns must be the same.

**Review by sriramyab — APPROVED** [review]
